src/file_to_json.py

```diff
@@ -1,63 +1,66 @@
-import pandas as pd
+import os
+import sys
+sys.path.append(os.path.abspath(''))
+import pandas as pd  # noqa: E402
 
 
-def main(blacklist=[]):
+def file_to_json(input_dir, output_dir, blacklist=[]):
     '''
     Reads excel/csv files into json format
     Excludes worksheets in blacklist
     '''
-    import os
-    json_ls = list()
-    f_walk = os.walk('data_folder')
+    f_walk = os.walk(input_dir)
 
-    for subdir, dir, files in f_walk:
+    for subdir, _, files in f_walk:
         for f in files:
             # pandas not only provides file reading functionality,
             # but outputs in a single format
             # if performance requires, other packages may be implemented,
             # that may require additional processing after reading in the file
             fp = os.path.join(subdir, f)
-            f_ext = f.split('.')[-1]
+            # using splitext allows '.' to appear in the filename
+            f_name, f_ext = os.path.splitext(f)
             table_ls = []
-            if f_ext[:3] == 'xls':
-                table = pd.read_excel(fp, sheet_name=None)
+            if f_ext[:4] == '.xls':
+                # openpyxl can open .xlsx but not .xls
+                # xlrd can open .xls but not .xlsx
+                if f_ext == '.xlsx':
+                    engine = 'openpyxl'
+                else:
+                    engine = 'xlrd'
+                table = pd.read_excel(fp, sheet_name=None, engine=engine)
```

Comment:
Pandas should determine which engine to use for read_excel based on the file extension

Reply:
Looks like omitting the `engine` argument will default to `xlrd` for both, even if `openpyxl` is installed.